### PR TITLE
Added a provision to add number of products to lists

### LINKED ISSUE #2251: Add a counter to products in list
Allow the user to add a number on products in lists. Ideally, the user could input fractions, so that it can work for:
I have 2 bottles of coke in my fridge
I have eaten 1/10 of the Nutella jar :-)

### PR DESCRIPTION
**Pull request by prashantkh19**
## Related issues and discussion
#fixes #2251 
 
 ## Screen-shots
 
device-2019-02-21-174839
device-2019-02-21-174909

### CODE REVIEW

**Comment by prashantkh19** [comment]
I am open to any UI suggestions.

**Comment by teolemon** [comment]
@prashantkh19 Do you need that much skin hydration ? 🤣

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -964,5 +964,9 @@
     <string name="settings_country_dialog_title">Select your country</string>
     <string name="settings_country_summary">Change your selected country</string>
     <string name="settings_country_title">Country</string>
+    <string name="txt_enter_count">Enter number of products:</string>
+    <string name="txt_hint_count">Input can be a faction also</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="txt_hint_count">Ex: 2, 1/2…</string>
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -964,5 +964,9 @@
...
+    <string name="txt_btn_count">Add</string>
+    <string name="toast_null_count">Number can\'t be null!</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="toast_null_count">This is not a valid number.</string>
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -964,5 +964,9 @@
     <string name="settings_country_dialog_title">Select your country</string>
     <string name="settings_country_summary">Change your selected country</string>
     <string name="settings_country_title">Country</string>
+    <string name="txt_enter_count">Enter number of products:</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="txt_enter_count">Enter the amount of product(s):</string>
```

**Comment by Karljoones** [comment]
Would it be possible to include the list name? 'Add to ....'

**Comment by prashantkh19** [comment]
> Would it be possible to include the list name? 'Add to ....'
I will do that.

**Comment by Karljoones** [comment]
Thank you!

**Comment by deniger** [comment]
warning: the database migration is not managed as a new column "count" is added.
Try to fix this...

**Comment by deniger** [comment]
To be added to DatabaseHelper.upgrade

case TheNewVersion: {
                if (!isFieldExist(db, "YOUR_LISTED_PRODUCT", "count")) {
                    db.execSQL(String.format("ALTER TABLE %s ADD COLUMN '%s' TEXT NOT NULL DEFAULT '';", "YOUR_LISTED_PRODUCT", "count"));
                }
                break;
            }


To be fixed:
+ https://github.com/openfoodfacts/openfoodfacts-androidapp/issues/2792
+ https://github.com/openfoodfacts/openfoodfacts-androidapp/issues/2793

**Pull request merged**